src/main.rs

```diff
@@ -127,6 +160,13 @@ fn main() {
         opts.metrics_addr.map(|_| metrics_sink),
     );
 
+    sub_command.map(|s| {
+        match s {
+            SubCommand::Validate(v) => validate(&v, &opts),
```

Comment:
a good idea might be for this to return a Result and when you get a error you return the config error code.

Reply:
I've refactored that section slightly to return the exit code itself as it might differ between CONFIG and USAGE.